xbmc/interfaces/builtins/PlayerBuiltins.cpp

```diff
@@ -474,6 +474,14 @@ int PlayOrQueueMedia(const std::vector<std::string>& params, bool forcePlay)
   // need some extended item properties to process resume successfully. Load them.
   item.LoadDetails();
 
+  // Check if video sources are locked - to capture recently added videos played from the home screen
+  if (VIDEO::IsVideo(item) && !g_passwordManager.IsVideoUnlocked())
```

Comment:
This method also handles music. Should a similar check for music be added?

Reply:
Music (including recently added) seems to be password checked before this point.

Reply:
Then i would at least try to find the place similar to the place where the check is done for music. Having the check here only for video feels not right.

Reply:
The code path for playing recently added items is very different for some reason:

![image](https://github.com/user-attachments/assets/94c8ad1c-64ce-42ff-bcd1-1fed83d2a634)

Playing from the movies menu:

![image](https://github.com/user-attachments/assets/81933fc2-f673-4358-af3f-b8e780333470)

There is nowhere in common with music and in the recently added play path, so it's in the best place I think.

Reply:
> so it's in the best place I think.

Sorry, no. There are definitely ways to call this for music!

<img width="944" alt="image" src="https://github.com/user-attachments/assets/f5594609-e017-4372-a643-2030912ddd57">

Example: Set default album select action to "Play" in estuary and click an album on music home screen recently added albums widget.

![image](https://github.com/user-attachments/assets/99ef8ef3-9f9f-4042-9d56-15d9a09b7781)

Reply:
My suggestion was not to find a different place to check videos but to add the check for music.

Reply:
And for reasons I wrote in my other comment, I would remove the code comment here completely.

Reply:
Never seen those settings before!! - you are right - recently added albums then follow the same path - updated.
Comments removed as per above.

Reply:
The code you are trying to fix/improve is definitely a beast. I dealt with it earlier but I'm still far away from claiming for myself that I really understood all of it. ;-)

Reply:
> The code you are trying to fix/improve is definitely a beast.

oh yeah, feels like profile manager is one big patchwork rug. i also had the pleasure